ksp_plugin/orbit_analyser.cpp

```diff
@@ -26,20 +28,14 @@ OrbitAnalyser::OrbitAnalyser(
       analysed_trajectory_parameters_(
           std::move(analysed_trajectory_parameters)) {}
 
-OrbitAnalyser::~OrbitAnalyser() {
-  if (analyser_.joinable()) {
-    keep_analysing_ = false;
-    analyser_.join();
-  }
-}
-
 void OrbitAnalyser::RequestAnalysis(
     Instant const& first_time,
     DegreesOfFreedom<Barycentric> const& first_degrees_of_freedom,
     Time const& mission_duration,
     not_null<RotatingBody<Barycentric> const*> primary) {
   if (!analyser_.joinable()) {
-    analyser_ = std::thread([this] { RepeatedlyAnalyseOrbit(); });
+    analyser_ = base::MakeStoppableThread(
```

Comment:
MakeStoppableThread should do the ignoring of the `stop_token` for us (we should never use it, since it should be accessed via `this_stoppable_thread::get_stop_token()` instead).